cloud/observability/promql-to-scrape/internal/server.go

```diff
@@ -0,0 +1,92 @@
+package internal
+
+import (
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"golang.org/x/exp/slog"
+)
+
+type PromToScrapeServer struct {
+	client             *APIClient
+	conf               *Config
+	server             http.Server
+	data               string
+	lastSuccessfulTime time.Time
+
+	sync.Mutex
+}
+
+func NewPromToScrapeServer(client *APIClient, conf *Config, addr string) *PromToScrapeServer {
+	s := &PromToScrapeServer{
+		client: client,
+		conf:   conf,
+		data:   "",
+	}
+	mux := http.NewServeMux()
+	mux.HandleFunc("/metrics", s.metricsHandler)
+	mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		http.Redirect(w, r, "/metrics", http.StatusMovedPermanently)
+	})
+
+	s.server = http.Server{
+		Addr:    addr,
+		Handler: mux,
+	}
+
+	go s.run()
+
+	return s
+}
+
+// metricsHandler is the HTTP handler for the "/metrics" endpoint.
+func (s *PromToScrapeServer) metricsHandler(w http.ResponseWriter, r *http.Request) {
+	s.Lock()
```

Comment:
A RWLock would be better here to allow multiple metricsHandlers to be serving at once but for our use cast it's not a major issue.